generator/src/main/java/com/scottlogic/deg/generator/generation/DecisionTreeDataGenerator.java

```diff
@@ -10,33 +11,45 @@
 import com.scottlogic.deg.generator.generation.databags.MultiplexingDataBagSource;
 import com.scottlogic.deg.generator.outputs.GeneratedObject;
 import com.scottlogic.deg.generator.walker.DecisionTreeWalker;
+import com.scottlogic.deg.generator.walker.DecisionTreeWalkerFactory;
+import com.scottlogic.deg.generator.walker.RuntimeDecisionTreeWalkerFactory;
+import com.scottlogic.deg.generator.walker.reductive.field_selection_strategy.FixFieldStrategy;
 
 import java.util.List;
 import java.util.stream.Collectors;
 import java.util.stream.Stream;
 
 public class DecisionTreeDataGenerator implements DataGenerator {
-    private final DecisionTreeWalker treeWalker;
+    private final GenerationConfig config;
     private final DataGeneratorMonitor monitor;
     private final TreePartitioner treePartitioner;
     private final DecisionTreeOptimiser treeOptimiser;
+    private final DecisionTreeFactory profileAnalyser;
+    private final FixFieldStrategy fixFieldStrategy;
 
     public DecisionTreeDataGenerator(
-            DecisionTreeWalker treeWalker,
-            TreePartitioner treePartitioner,
-            DecisionTreeOptimiser optimiser,
-            DataGeneratorMonitor monitor) {
+        GenerationConfig config,
+        TreePartitioner treePartitioner,
+        DecisionTreeOptimiser optimiser,
+        DataGeneratorMonitor monitor,
+        DecisionTreeFactory profileAnalyser,
+        FixFieldStrategy fixFieldStrategy) {
+        this.config = config;
         this.treePartitioner = treePartitioner;
         this.treeOptimiser = optimiser;
-        this.treeWalker = treeWalker;
         this.monitor = monitor;
+        this.profileAnalyser  = profileAnalyser;
+        this.fixFieldStrategy  = fixFieldStrategy;
     }
 
     @Override
     public Stream<GeneratedObject> generateData(
         Profile profile,
-        DecisionTree decisionTree,
         GenerationConfig generationConfig) {
+        DecisionTree decisionTree = profileAnalyser.analyse(profile).getMergedTree();
```

Comment:
I'm not convinced this is right.
This type has gone from a "generator of data given a tree" to a "profile analyser and strategy selection generator of data based on the runtime configuration". Is there a way to change this back to focus its responsibility to that of generating data given a tree rather than overloading it with all the other responsibilities.